application/server/server.go

```diff
@@ -547,21 +547,28 @@ func textDocumentDidSaveHandler() jrpc2.Handler {
 		bgCtx := context.Background()
 		c := config.CurrentConfig()
 		logger := c.Logger().With().Str("method", "TextDocumentDidSaveHandler").Logger()
-
 		logger.Info().Interface("params", params).Msg("Receiving")
+
+		autoScanEnabled := c.IsAutoScanEnabled()
+
 		di.FileWatcher().SetFileAsSaved(params.TextDocument.URI)
 		filePath := uri.PathFromUri(params.TextDocument.URI)
 
-		f := workspace.Get().GetFolderContaining(filePath)
-		autoScanEnabled := config.CurrentConfig().IsAutoScanEnabled()
-		if f != nil && autoScanEnabled {
-			go f.ScanFile(bgCtx, filePath)
-		} else {
+		ws := workspace.Get()
+		if ws != nil && autoScanEnabled && uri.IsDotSnykFile(params.TextDocument.URI) {
+			go ws.ScanWorkspace(bgCtx)
```

Comment:
Question: why not scan the workspaceFolder? Scanning the workspace could be too much - especially in Eclipse, corporate users often have 30+ projects (=workspace folders) open, and scanning all of them may slow down their workflow & computer. You could pull line 563 up, and then call f.Scan(). WDYT?